app/src/main/java/com/example/giphyservice/ui/main/MainViewModel.kt

```diff
@@ -23,14 +26,22 @@ class MainViewModel(private val gifsRepository: GifsRepository) : ViewModel() {
     fun loadData() {
         state.value = UIState.Loading
 
-        gifsRepository.getGifsData(object : GifCallback {
-            override fun onSuccess(gifs: List<Gif>) {
-                state.value = UIState.Success(gifs)
+        viewModelScope.launch(Dispatchers.IO) {
+            try {
+                val result = gifsRepository.getGifsData()
+                withContext(Dispatchers.Main) {
```

Comment:
you can update the state in any thread, it doesn't matter. The only think matters is in which thread you observe it